src/python/pants/jvm/strip_jar/strip_jar.py

```diff
@@ -0,0 +1,201 @@
+        ),
+    )
+
+    return FallibleStripJarResult(process_result=process_result)
+
+
+@rule(level=LogLevel.DEBUG)
+async def resolve_fallible_result_to_strip_jar(
+    fallible_result: FallibleStripJarResult,
+    process_cleanup: ProcessCleanupOption,
+) -> Digest:
+    if fallible_result.process_result.exit_code == 0:
+        digest = await Get(
+            Digest, RemovePrefix(fallible_result.process_result.output_digest, _OUTPUT_PATH)
+        )
+        return digest
+    raise ProcessExecutionFailure(
+        fallible_result.process_result.exit_code,
+        fallible_result.process_result.stdout,
+        fallible_result.process_result.stderr,
+        "Strip jar failed.",
+        process_cleanup=process_cleanup.val,
+    )
+
+
+def _load_strip_jar_source() -> bytes:
+    return pkg_resources.resource_string(__name__, _STRIP_JAR_BASENAME)
+
+
+@rule
+async def build_processors(jdk: InternalJdk) -> StripJarCompiledClassfiles:
+    dest_dir = "classfiles"
+    lockfile_request = await Get(GenerateJvmLockfileFromTool, StripJarToolLockfileSentinel())
+    materialized_classpath, source_digest = await MultiGet(
+        Get(
+            ToolClasspath,
+            ToolClasspathRequest(prefix="__toolcp", lockfile=lockfile_request),
+        ),
+        Get(
+            Digest,
+            CreateDigest(
+                [
+                    FileContent(
+                        path=_STRIP_JAR_BASENAME,
+                        content=_load_strip_jar_source(),
+                    ),
+                    Directory(dest_dir),
+                ]
+            ),
+        ),
+    )
+
+    merged_digest = await Get(
+        Digest,
+        MergeDigests(
+            (
+                materialized_classpath.digest,
+                source_digest,
+            )
+        ),
+    )
+
+    process_result = await Get(
+        ProcessResult,
+        JvmProcess(
+            jdk=jdk,
+            classpath_entries=[f"{jdk.java_home}/lib/tools.jar"],
+            argv=[
+                "com.sun.tools.javac.Main",
+                "-cp",
+                ":".join(materialized_classpath.classpath_entries()),
+                "-d",
+                dest_dir,
+                _STRIP_JAR_BASENAME,
+            ],
+            input_digest=merged_digest,
+            output_directories=(dest_dir,),
+            description=f"Compile {_STRIP_JAR_BASENAME} import processors with javac",
+            level=LogLevel.DEBUG,
+            # NB: We do not use nailgun for this process, since it is launched exactly once.
+            use_nailgun=False,
+        ),
+    )
+    stripped_classfiles_digest = await Get(
+        Digest, RemovePrefix(process_result.output_digest, dest_dir)
+    )
+    return StripJarCompiledClassfiles(digest=stripped_classfiles_digest)
+
+
+@rule
+def generate_strip_jar_lockfile_request(
+    _: StripJarToolLockfileSentinel,
+) -> GenerateJvmLockfileFromTool:
+    return GenerateJvmLockfileFromTool(
```

Comment:
Should the user be able to change the tool version? (Which would favor this rule using `GenerateJvmLockfileFromTool.create` with a `JvmToolBase` subclass.

Reply:
do you have an example I can use for this? I'm not familiar with this part.

Reply:
Just look for `JvmToolBase` subclasses. `git grep '^class.*JvmToolBase'` should reveal them.

Reply:
(JvmToolBase itself subclasses `Subsystem` and so exposes user-visible options for changing the version of the tool. You then hook your subclass into the generate-lockfiles machinery by using `GenerateJvmLockfileFromTool.create` on an instance of that subclass (which can just be a parameter to your rule that the engine fills in).

Reply:
got it, thanks

Reply:
@tdyas I'm not it worth in this case, the last commit to the tool was in 2018:

https://github.com/raboof/reproducible-builds-jvm-stripper

The tool is pretty stable and used by sbt-reproducible-builds:

https://github.com/raboof/sbt-reproducible-builds

Reply:
👍